### PR TITLE
v1.13.0 data_fusion - VERSION RELEASE - reader kwargs

### LINKED ISSUE #629: v1.12.4 - v1.13.0 Open Source Release
Contents of this Issue will be updated throughout the release process.
  If this text still exists, the release is not yet complete.

# Repositories tagged with 1.13.0
* ancildat
* data_fusion
* geocolor
* geoips
* geoips_clavrx
* geoips_plugin_example
* recenter_tc
* template_test_data
* test_data_tpw
* test_data_tpw_1.12.4_github.com.tgz
* test_data_viirs
* test_data_viirs_1.12.4_github.com.tgz

# Testing instructions for 1.13.0 release

* Install GeoIPS ( https://github.com/NRLMMD-GEOIPS/geoips#installation )
  * You do not have to reinstall, but follow appropriate installation instructions if you do not have a current install
* Enable GeoIPS environment
  * Activate the Python environment containing GeoIPS installation.
  * Ensure all required development environment variables are set
    * $GEOIPS_PACKAGES_DIR, $GEOIPS_REPO_URL, $GEOIPS_TESTDATA_DIR, $GEOIPS_OUTDIRS)
* Switch to v1.13.0-release branch for geoips repo

```
git -C $GEOIPS_PACKAGES_DIR/geoips pull
git -C $GEOIPS_PACKAGES_DIR/geoips checkout v1.13.0-release
```
* Get all of the geoips plugin packages and test data repos.
  Note the full install uses > 60G disk space and > 16G memory.
```

# Just the geoips repo
$GEOIPS_PACKAGES_DIR/geoips/tests/integration_tests/base_install.sh

# Make sure no unit tests, etc fail when ONLY the geoips repo is installed.
# This only gives us a good test if you are working from a brand new install.
$GEOIPS_PACKAGES_DIR/geoips/tests/utils/check_code.sh all $GEOIPS_PACKAGES_DIR/geoips

# All plugin packages and test data (and cartopy shapefiles)
export CARTOPY_DATA_DIR=$HOME/geoips/cartopy_data_dir                                                                                                           
$GEOIPS_PACKAGES_DIR/geoips/tests/integration_tests/full_install.sh
```
* Follow the below instructions to switch any repos with a release branch to that branch:
```
# convenience functions to run on all packages in $GEOIPS_PACKAGES_DIR
source $GEOIPS_PACKAGES_DIR/geoips/setup/bash_setup/gitall
gitall pull  # make sure we have all the branches, etc available
gitall checkout main  # switch everything to main
# this will switch anything with release branch there, and do nothing otherwise
gitall checkout v1.13.0-release
gitall pull  # make sure everything is up to date
pipall install -e .  # now install
create_plugin_registries  # create the geoips plugin registries
```
* Run the base test, which will ensure the geoips base repository updates
  were successful:
```
$GEOIPS_PACKAGES_DIR/geoips/tests/integration_tests/base_test.sh
```
* Run the full test, which will ensure ALL GeoIPS plugin repositories were
  updated successfully (This will use a large amount of disk space and
  memory!  If you are memory or disk space limited, only perform the base
  install and test)
```
$GEOIPS_PACKAGES_DIR/geoips/tests/integration_tests/full_test.sh
```

## Test outputs for 1.13.0 release

Base test:

```
Package: geoips_base
Total run time: 175 seconds
Number data types run: 3
Number data types failed: 0
Tue Jun 18 04:31:45 UTC 2024
```

Full test had 2 failures

```
Package: geoips_full
Total run time: 3440 seconds
Number data types run: 58
Number data types failed: 2
Tue Jun 18 14:18:18 UTC 2024
```

template_fusion_plugin had a failure, but we are considering getting rid of it in favor of template_basic_plugin + geoips_plugin_example.

```
Tue Jun 18 07:05:52 PDT 2024  Running test_all.sh
/users/surratt/geoips/template_fusion_plugin/tests/test_all.sh
/users/surratt/geoips/outdirs/logs/20240618/20240618.132058_geoips_full/test_all_geoips_full.log_test_all.sh.log
        Return: 1
```

One failure on full_test, Evan is working on it:

image

Also a failure on pytest_short when full_install has not been run (when recenter_tc is not installed, so right after a clean install when base_install/base_test have been run, but full_install/full_test have not)

image

### PR DESCRIPTION
**Pull request by mindyls**
# Related Issues
required to close NRLMMD-GEOIPS/geoips#629
required to close NRLMMD-GEOIPS/geoips#629

# Reviewer Instructions
* Please confirm this PR is set up to merge from v1.13.0-release branch into main branch
* Review "Files changed" tab
* Confirm appropriate testing was completed for these updates (you may perform tests yourself, or review output included from another reviewer/assignee)

# Summary
Changes for version 1.13.0, uploaded with
1.13.0 update on repo data_fusion.
See release note updates in 'Files changed' tab

See NRLMMD-GEOIPS/geoips#629 for additional repositories included in this update.
See NRLMMD-GEOIPS/geoips#629 for dev updates included in this update.

# Testing Instructions
See NRLMMD-GEOIPS/geoips#629 for testing instructions.

# Output
See NRLMMD-GEOIPS/geoips#629 for test output.

# Post Merge Steps
Once this PR has been merged, v1.13.0
will be tagged/released on the data_fusion main branch.

### CODE REVIEW

File: data_fusion/plugins/yaml/products/stitched.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@ spec:
             arguments: {}
         colormapper:
           plugin:
-            name: linear_norm
+            name: matplotlib_linear_norm
```
Comment (mindyls) [inline data_fusion/plugins/yaml/products/stitched.yaml:25]: @evrose54  we need to make sure this gets updated back to linear_norm once the geoips PR has been merged.

**Review by jsolbrig — APPROVED** [review]

File: data_fusion/plugins/yaml/products/stitched.yaml (hunk 1 above)
Comment (evrose54) [reply data_fusion/plugins/yaml/products/stitched.yaml:25]: Sounds good. Once [NRLMMD/geoips#504](https://github.com/NRLMMD-GEOIPS/geoips/pull/504) is merged we can make the change back to linear norm.

File: data_fusion/plugins/modules/output_formatters/layered_imagery.py [hunk 2]
```diff
@@ -407,8 +407,8 @@ def call(
         mapobj,
         main_ax,
         area_def,
-        feature_annotator=boundaries_info,
-        gridline_annotator=gridlines_info,
+        feature_annotator=feature_annotator,
+        gridline_annotator=gridline_annotator,
```
Comment (christopher-selman) [inline data_fusion/plugins/modules/output_formatters/layered_imagery.py:411]: Is this hedging against a particular scenario or was there an issue encountered with just passing feature_annotator as an argument? What I'm getting at is: is there any harm in just having it be 
```
feature_annotator,
gridline_annotator,
```

**Review by christopher-selman — APPROVED** [review]

File: data_fusion/plugins/modules/output_formatters/layered_imagery.py (hunk 2 above)
Comment (mindyls) [reply data_fusion/plugins/modules/output_formatters/layered_imagery.py:411]: I tend to pass in arg=var when it is a kwarg.  I don't think there is any harm either way?

**Review by mindyls — COMMENTED** [review]
image

pytest error is in geoips repo - should be resolved with future merges.

image

release note 1.13.0 IS modified, so going to ignroe release note update error (checking for 1.14.0)

image

I looked at every file!

Merging without requirements being met.

**Pull request merged**